src/compiler/checker.ts

```diff
@@ -23258,6 +23263,9 @@ namespace ts {
 
         function checkExportSpecifier(node: ExportSpecifier) {
             checkAliasSymbol(node);
+            if (compilerOptions.declaration) {
+                collectLinkedAliases(node.propertyName || node.name, /*setVisibility*/ true); // Collect linked aliases to set visibility links

```

Comment:
both this comment and the other one seem redundant. And if they're not, then things are actually *more* complicated than they appear.

Reply:
They didn't feel redundant before I added the `setVisibility` flag. :(

I'll open a PR to remove them.